### PR TITLE
fix(sidebar): fix header not respecting container width - FE-4232

### LINKED ISSUE #4240: [ Sidebar ] header doesn't respect the container width when rendered within a sidebar
### Current behaviour

The Header in sidebar expands more than the width of the sidebar.

Screenshot 2021-07-14 at 15 13 46

### Expected behaviour

The Header in sidebar remains the same width of the sidebar.

### Reproducible example

https://codesandbox.io/s/white-sunset-vqypu?file=/src/index.js

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| carbon-react     | 77.12.1 |
| carbon-factory   |
| react-scripts    | 
| React            |
| Browser          | All |
| npm              |
| Operating System | MacOS |

### PR DESCRIPTION
**Pull request by DipperTheDan**
fixes the sidebar header not respecting the container width when left aligned

fixes #4240

### Proposed behaviour
Screenshot 2021-07-22 at 11 06 17

### Current behaviour
Screenshot 2021-07-22 at 11 06 51

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [ ] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Additional context
Screenshots look different widths because they are. The `Proposed behaviour` screenshot uses the default size where as the codesandbox and `Current behaviour` screenshot uses the `small` size.

### Testing instructions
- Created a new story called `with Typography`
- Added a new knob for Test story for `Header` so future manual testing can be completed in the `Header`
- Check issue codesandbox behaves as expected. (https://codesandbox.io/s/mystifying-brook-n712k)

### CODE REVIEW

File: src/components/sidebar/__internal__/sidebar-header/sidebar-header.style.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ const SidebarHeaderStyle = styled.div`
   background-color: ${({ theme }) => theme.colors.white};
   box-shadow: inset 0 -1px 0 0 ${({ theme }) => theme.disabled.border};
   box-sizing: content-box;
-  width: 100%;
+  width: 100% - 32px;
```
Comment (sjohnson-sage) [inline src/components/sidebar/__internal__/sidebar-header/sidebar-header.style.js:8]: I don't think this is really fixing the problem, just covering it up with specific CSS

I think the `box-sizing: content-box;` is the issue, and it should be changed to `box-sizing: border-box;` to take the padding into account

https://developer.mozilla.org/en-US/docs/Web/CSS/box-sizing

**Review by sjohnson-sage — CHANGES_REQUESTED** [review]

File: src/components/sidebar/__internal__/sidebar-header/sidebar-header.style.js (hunk 1 above)
Comment (DipperTheDan) [reply src/components/sidebar/__internal__/sidebar-header/sidebar-header.style.js]: Good point 😃

File: src/components/sidebar/sidebar.stories.js [hunk 2]
```diff
@@ -52,6 +55,7 @@ export const Default = () => {
         position={position}
         size={size}
         onCancel={onCancel}
+        header={headerChildren.length > 0 ? headerChildren : null}
```
Comment (sjohnson-sage) [inline src/components/sidebar/sidebar.stories.js:58]:
```suggestion
        header={headerChildren}
```

No need for the extra bits 👍🏻

Comment (DipperTheDan) [reply src/components/sidebar/sidebar.stories.js:58]: Consider it gone.

**Review by sjohnson-sage — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**